.github/workflows/node.yml

```diff
@@ -90,27 +96,25 @@ jobs:
             exit 1
           fi
 
-  test-ts:
+  test:
     runs-on: ubuntu-latest
     needs: [yarn-build]
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
+          fetch-depth: 0
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
 
       - uses: actions/cache@v3
         with:
           path: ./*
           key: ${{ github.sha }}
 
-      - name: sdk
-        run: yarn workspace @hyperlane-xyz/sdk run test
-
-      - name: helloworld
-        run: yarn workspace @hyperlane-xyz/helloworld run test
-
-      - name: infra
-        run: yarn workspace @hyperlane-xyz/infra run test
+      - name: Unit Tests
+        run: yarn test
```

Comment:
I don't see where the magic happens to prevent test-sol when no changes were made there. Is that part of this?

Reply:
the `--since` flag in the root `workspaces foreach` commands does this, which this leverages